src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -657,9 +654,11 @@ void Assembler::emit_operand_helper(int reg_enc, int base_enc, int index_enc,
       }
     } else {
       // [base + disp]
-      assert(base_enc != rsp->encoding() LP64_ONLY(&& base_enc != r12->encoding()), "illegal addressing mode");
-      if (disp == 0 && no_relocation &&
-          base_enc != rbp->encoding() LP64_ONLY(&& base_enc != r13->encoding())) {
+      assert(base_enc != rsp->encoding() LP64_ONLY(&& base_enc != r12->encoding()
+             && base_enc != 20 && base_enc != 28), "illegal addressing mode");
+      if (disp == 0 && no_relocation
+          && base_enc != rbp->encoding() LP64_ONLY(&& base_enc != r13->encoding()
+          && base_enc != 21 && base_enc != 29)) {
```

Comment:
// Note: rsp->encoding() is 4 and rbp->encoding() is 5
      // base_enc !=  4, 12, 20, 28
      assert(((base_enc & 0x7) != 4), "illegal addressing mode");
      // base_enc is one of  5, 13, 21, 29
      if (disp == 0 && no_relocation &&  ((base_enc & 0x7) != 5)) {

Reply:
Thanks. Done.